dvc/fs/utils.py

```diff
@@ -1,75 +1,99 @@
 import errno
 import logging
 import os
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, List, Optional
 
 from .base import RemoteActionNotImplemented
 from .local import LocalFileSystem
 
 if TYPE_CHECKING:
-    from dvc.types import AnyPath
-
     from .base import BaseFileSystem
+    from .types import AnyPath
 
 logger = logging.getLogger(__name__)
 
 
 def _link(
+    link: "AnyPath",
     from_fs: "BaseFileSystem",
```

Comment:
Will rip out `AnyPath` and rename `BaseFileSystem` -> `FileSystem` in the followup. Just keeping this for consistency for now.